path_test.go

```diff
@@ -125,6 +125,14 @@ func Test_zoneFromPath(t *testing.T) {
 			"_redirect.string.example.com",
 			nil,
 		},
+		{
+			"example.com",
+			"/12345-some-path?query=string&more=stuff",
+			"",
+			"\\?query=(?P<a>[^&]+)\\&more=(?P<b>[^&]+)",
```

Comment:
We should do a more complicated test that actually reorders the matches.

``` suggestion
			"\\?query=(?P<b>[^&]+)\\&more=(?P<a>[^&]+)\\&test=(?P<c>[^&]+)",
```

Additionally we should test for the correct handling of some errors such as using the same ordered match multiple times.

Reply:
Will add more tests and error handlings.